gateway/ln-gateway/src/events.rs

```diff
@@ -0,0 +1,51 @@
+use fedimint_core::core::ModuleKind;
+use fedimint_core::Amount;
+use fedimint_eventlog::{Event, EventKind};
+use fedimint_lnv2_common::contracts::{Commitment, OutgoingContract};
+use serde::{Deserialize, Serialize};
+
+use crate::sha256::Hash;
+
+#[derive(Serialize, Deserialize)]
+pub struct OutgoingPaymentSuccess {
+    pub outgoing_contract: OutgoingContract,
+    pub min_contract_amount: Amount,
+    pub invoice_amount: Amount,
+    pub start_millis: u128,
+    pub end_millis: u128,
+    pub max_delay: u64,
+    pub is_swap: bool,
+}
+
+impl Event for OutgoingPaymentSuccess {
+    const MODULE: Option<ModuleKind> = Some(fedimint_lnv2_common::KIND);
+
+    const KIND: EventKind = EventKind::from_static("outgoing-payment-success");
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct IncomingPaymentSuccess {
+    pub incoming_contract_commitment: Commitment,
+    pub amount: Amount,
+    pub start_millis: u128,
```

Comment:
Json has numerical precision issues, so I'm not sure if there's a point of using `u128`

Reply:
https://github.com/fedimint/fedimint/blob/33efb3408b2b08426da7dd9caf51e1efa164e738/fedimint-eventlog/src/lib.rs#L157

Reply:
Using `millis` in rust returns a `u128`. What are the precision issues? Would using a `String` or something be better?

Reply:
https://www.pullrequest.com/blog/safely-handling-large-integers-in-json-best-practices-and-pitfalls/ , JS numbers are floats, and JSON is fundamnetally a JS-based interoperability format.

`millis` return `u128`, but for real timestamps `u64` is enough for next thousands of years.

Reply:
I can use a `u64`